src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/HmppsDomainEventEmitter.kt

```diff
@@ -120,3 +163,24 @@ data class PrisonerCreatedDomainEvent(
       detailUrl = ServletUriComponentsBuilder.fromUriString(host).path("/prisoner/{offenderNo}").buildAndExpand(additionalInfo.nomsNumber).toUri().toString(),
     )
 }
+
+data class PrisonerReceivedEvent(val nomsNumber: String, val reason: String, val prisonId: String, val fromPrisonId: String?)
+
+data class PrisonerReceivedDomainEvent(
+  val additionalInfo: PrisonerReceivedEvent,
+  val occurredAt: String,
+  val eventType: String,
+  val version: Int,
+  val description: String,
+  val detailUrl: String,
+) {
+  constructor(additionalInfo: PrisonerReceivedEvent, occurredAt: Instant, host: String) :
+    this(
+      additionalInfo = additionalInfo,
+      occurredAt = ISO_OFFSET_DATE_TIME.withZone(ZoneId.of("Europe/London")).format(occurredAt),
+      eventType = PRISONER_RECEIVED_EVENT_TYPE,
+      version = 1,
+      description = "A prisoner has been received into a prison",
+      detailUrl = ServletUriComponentsBuilder.fromUriString(host).path("/prisoner/{offenderNo}").buildAndExpand(additionalInfo.nomsNumber).toUri().toString(),
+    )
```

Comment:
I'm guessing that we'll end up with a single `DomainEvent<E>` type once we have a few of these.

Reply:
100%